app/src/main/java/sdp/moneyrun/game/Game.java

```diff
@@ -90,6 +90,56 @@ public Game(String name,
         started = false;
     }
 
+    public Game(String name,
+                Player host,
+                int maxPlayerCount,
+                List<Riddle> riddles,
+                List<Coin> coins,
+                Location startLocation,
+                boolean isVisible,
+                int numCoins,
+                double radius,
+                double duration) {
+        if (name == null) {
+            throw new IllegalArgumentException("name should not be null.");
+        }
+        if (host == null) {
+            throw new IllegalArgumentException("host should not be null.");
+        }
+        if (riddles == null) {
+            throw new IllegalArgumentException("riddles should not be null.");
+        }
+        if (coins == null) {
+            throw new IllegalArgumentException("coins should not be null.");
+        }
+        if (startLocation == null) {
+            throw new IllegalArgumentException("startLocation should not be null.");
+        }
+        if (maxPlayerCount <= 0) {
+            throw new IllegalArgumentException("maxPlayerCount should not be smaller than 1.");
+        }
+
+        if (numCoins <= 0) {
+            throw new IllegalArgumentException("Number of coins should be bigger than 0.");
+        }
+        if (radius <=0 ) {
+            throw new IllegalArgumentException("Radius should be bigger than 0.");
+        }
+        if (duration <=0) {
+            throw new IllegalArgumentException("Duration should be bigger than 0.");
+        }
+
+        this.id = null;
```

Comment:
I think initializing a variable to null is useless, if a variable is not initialized it is already null.

Reply:
Yes